src/data/storage-area.cpp

```diff
@@ -56,90 +57,30 @@ const fs::path GetPath(OrthancPluginContentType type, const char* uuid){
-                // todo: probably a better error code
-                return OrthancPluginErrorCode_EmptyRequest;
-            }
-            // write to disk
-            fs::create_directories(master_path.parent_path());
-
-            sprintf(msg,"writing file: %s", master_path.string().c_str());
-            DEBUG_LOG(0,msg)
-            std::fstream file(master_path, std::ios::binary | std::ios::out);
-            file.write(content.get(),size);
-            file.close();
-        } else {
-            DEBUG_LOG(1,"Nothing was filtered");
-            dicom.Write(uuid);
-        }
-        // set file permissions
-        // todo: permission debug info?
-        fs::file_status master_status = fs::status(master_path);
-        master_status.permissions(globals::file_permissions);
-        DEBUG_LOG(1,"WriteDicomFile: permissions set");
-        // create hard links
-        auto hardlink_to = [&](std::string groupby, std::string group) {
-            fs::path link = fs::path(storage_root)
-                    .append(groupby)
-                    .append(group)
-                    .append(uuid)
-                    .append(".DCM");
-            fs::create_directories(link);
-            fs::create_hard_link(master_path, link);
-            fs::permissions(link, globals::file_permissions);
-        };
-        // todo: integrate json settings to enable/disable individual hard links
-        // todo: replace placeholders
-        if(false) {
-            std::string DOB_placeholder;
-            std::string PID_placeholder;
-            std::string SD_placeholder;
-            try {
-                hardlink_to("/by-dob/", DOB_placeholder);
-                hardlink_to("/by-patient-id/", PID_placeholder);
-                hardlink_to("/by-study-date/", SD_placeholder);
-            } catch (const std::exception &e) {
-                DEBUG_LOG(PLUGIN_ERRORS,"We failed to create hard links. They may already exist. OR the placeholders still aren't replaced.")
-                std::cerr << e.what() << std::endl;
-            }
-        }
-        DEBUG_LOG(1,"WriteDicomFile: success");
-        return OrthancPluginErrorCode_Success;
-    }
-    return OrthancPluginErrorCode_BadFileFormat;
-}
-
 OrthancPluginErrorCode StorageCreateCallback(const char *uuid,
                                              const void *content,
                                              int64_t size,
                                              OrthancPluginContentType type) {
-    fs::path path;
+    char msg[1024] = {0};
+    sprintf(msg,"StorageCreateCallback:\nuuid: %s\ncontent: %zu\n", uuid, content);
+    DEBUG_LOG(1,msg)
+    fs::path path = GetPath(type, uuid);
     switch (type) {
-        case OrthancPluginContentType_Dicom:
-            return WriteDicomFile(DicomFile(content, size), uuid);
+        case OrthancPluginContentType_Dicom: {
+            DicomFile file(content, size);
```

Comment:
We should check if the file is valid before passing it to the anonymizer.

Reply:
I guess Anonymize() already does that

Reply:
It does, part of the reasoning why it returns bool.